orderer/common/cluster/util.go

```diff
@@ -944,3 +922,55 @@ func SHA256Digest(data []byte) []byte {
 	hash := sha256.Sum256(data)
 	return hash[:]
 }
+
+// VerifyBlocksBFT verifies the given consecutive sequence of blocks is valid, always verifies signature,
+// and returns nil if it's valid, else an error.
+func VerifyBlocksBFT(blocks []*common.Block, signatureVerifier protoutil.BlockVerifierFunc) error {
+	return verifyBlockSequence(blocks, signatureVerifier, true)
+}
+
+func verifyBlockSequence(blockBuff []*common.Block, signatureVerifier protoutil.BlockVerifierFunc, alwaysCheckSig bool) error {
```

Comment:
This is only used now for BFT, no?

So we don't need this complex logic. Just verify the signature of every block, and no need to check the hash chain anymore if the signature verifies.